backend/tests/unit/controllers/posts-controller.test.js

```diff
@@ -0,0 +1,424 @@
+import {
+  createPostHandler,
+  getAllPostsHandler,
+  getFeaturedPostsHandler,
+  getPostByCategoryHandler,
+  getLatestPostsHandler,
+  getPostByIdHandler,
+  updatePostHandler,
+  deletePostByIdHandler,
+} from '../../../controllers/posts-controller.js';
+import Post from '../../../models/post.js';
+import { validCategories } from '../../../utils/constants.js';
+
+jest.mock('../../../models/post.js', () => ({
+  __esModule: true,
+  default: jest.fn(),
+}));
+
+const res = {
+  json: jest.fn(),
+  status: jest.fn().mockReturnThis(),
+};
+
+describe('createPostHandler', () => {
+  it('should create a new post', async () => {
+    const postObject = {
+      title: 'Test Post',
+      authorName: 'Test Author',
+      imageLink: 'https://www.forTestingPurposeOnly/my-image.jpg',
+      categories: [validCategories[0]],
+      description: 'This is a test post.',
+      isFeaturedPost: false,
+    };
```

Comment:
Can we create a private function which returns us the Post which takes in the attributes via params and use it everywhere, right now it is too much spread around the place.